### ISSUE TITLE
panic in buildTableReaderForIndexJoin

### ISSUE THREAD

**Issue by wjhuang2016** [pingcap/tidb#40596]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

``` SQL
CREATE TABLE `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd` (
  `791be5bd-3a41-434f-b194-0a8f7f5fa27b` double DEFAULT '1.335088259490289',
  `cb64194a-f141-4bfb-a2dd-dd78b6cf9f91` set('mj','4s7ht','z','3i','b26','9','cg11','uvzcp','c','ns','fl9') NOT NULL DEFAULT 'mj,z,3i,9,cg11,c',
  PRIMARY KEY (`cb64194a-f141-4bfb-a2dd-dd78b6cf9f91`) /*T![clustered_index] CLUSTERED */,
  KEY `013a92c6-db30-4963-b902-d4943d4023ed` (`791be5bd-3a41-434f-b194-0a8f7f5fa27b`),
  KEY `7f8209dc-6f2e-4646-9ddf-51c62e48b6dc` (`791be5bd-3a41-434f-b194-0a8f7f5fa27b`),
  KEY `2347bbe5-fe5a-4984-b2c5-79f451095482` (`791be5bd-3a41-434f-b194-0a8f7f5fa27b`)
) ENGINE=InnoDB DEFAULT CHARSET=gbk COLLATE=gbk_chinese_ci;

INSERT INTO `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd` VALUES (634.2783557491367,''),(2000.5041449792013,'4s7ht'),(634.2783557491367,'3i'),(634.2783557491367,'9'),(7803.173688589342,'uvzcp'),(634.2783557491367,'ns'),(634.2783557491367,'fl9');

CREATE TABLE `9aed23b9-9772-40dc-951c-7263c605c776` (
  `a480d6d7-02aa-48d8-b8ad-fffb94689039` decimal(56,16) DEFAULT '931359772706767457132645278260455518957.9866038319986886',
  `f8c37c8b-9167-4f46-a378-c1863f858433` set('3bqx','g','6op3','2g','jf','arkd3','y0b','jdy','1g','ff5z','224b') DEFAULT '3bqx,2g,ff5z,224b',
  `c0083600-1ec7-4176-bbe2-b8a4a71596d7` smallint(6) NOT NULL DEFAULT '-25973',
  `e542dbe8-59c2-4ec8-bec0-c179fea8f542` year(4) DEFAULT '2122',
  `90c28415-53f7-4443-b5e1-4177e42f0f39` text DEFAULT NULL,
  PRIMARY KEY (`c0083600-1ec7-4176-bbe2-b8a4a71596d7`) /*T![clustered_index] CLUSTERED */,
  KEY `d20dfca1-6dda-4e65-a4f2-c0e3f5cb1b2a` (`e542dbe8-59c2-4ec8-bec0-c179fea8f542`),
  KEY `482006a0-2c85-45b1-9c4a-3a94c8f12ded` (`c0083600-1ec7-4176-bbe2-b8a4a71596d7`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin COMMENT='bbbf3f9c-a23e-4607-979a-0d4ce886f2c0'
PARTITION BY HASH (`c0083600-1ec7-4176-bbe2-b8a4a71596d7`) PARTITIONS 4;

INSERT INTO `9aed23b9-9772-40dc-951c-7263c605c776` VALUES (465.0000000000000000,'jdy',-8542,2008,'FgZXe');

select    /*+ inl_join( `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd` , `9aed23b9-9772-40dc-951c-7263c605c776` ) */ avg(   `9aed23b9-9772-40dc-951c-7263c605c776`.`c0083600-1ec7-4176-bbe2-b8a4a71596d7` ) as r0 , repeat( `9aed23b9-9772-40dc-951c-7263c605c776`.`90c28415-53f7-4443-b5e1-4177e42f0f39` , `9aed23b9-9772-40dc-951c-7263c605c776`.`c0083600-1ec7-4176-bbe2-b8a4a71596d7` ) as r1 , locate( `9aed23b9-9772-40dc-951c-7263c605c776`.`90c28415-53f7-4443-b5e1-4177e42f0f39` , `9aed23b9-9772-40dc-951c-7263c605c776`.`90c28415-53f7-4443-b5e1-4177e42f0f39` ) as r2 , unhex( `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd`.`791be5bd-3a41-434f-b194-0a8f7f5fa27b` ) as r3 from `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd` right join `9aed23b9-9772-40dc-951c-7263c605c776` on `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd`.`cb64194a-f141-4bfb-a2dd-dd78b6cf9f91` = `9aed23b9-9772-40dc-951c-7263c605c776`.`c0083600-1ec7-4176-bbe2-b8a4a71596d7` where not( `9aed23b9-9772-40dc-951c-7263c605c776`.`c0083600-1ec7-4176-bbe2-b8a4a71596d7` in ( -7860 ,-13384 ,-12940 ) ) and not( `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd`.`cb64194a-f141-4bfb-a2dd-dd78b6cf9f91` between '4s7ht' and 'mj' );
```

### 2. What did you expect to see? (Required)
No error

### 3. What did you see instead (Required)

```
[index_lookup_join.go:484] ["innerWorker panicked"] [conn=3533669842974415417] [recover="runtime error: index out of range [2] with length 2"] [stack="github.com/pingcap/tidb/executor.(*innerWorker).run.func1\n\t/Users/pingcap/workspace/build-common@4/go/src/github.com/pingcap/tidb/executor/index_lookup_join.go:484\nruntime.gopanic\n\t/usr/local/go1.19.3/src/runtime/panic.go:884\nruntime.goPanicIndex\n\t/usr/local/go1.19.3/src/runtime/panic.go:113\ngithub.com/pingcap/tidb/table/tables.(*partitionedTable).locateHashPartition\n\t/Users/pingcap/workspace/build-common@4/go/src/github.com/pingcap/tidb/table/tables/partition.go:1112\ngithub.com/pingcap/tidb/table/tables.(*partitionedTable).locatePartition\n\t/Users/pingcap/workspace/build-common@4/go/src/github.com/pingcap/tidb/table/tables/partition.go:994\ngithub.com/pingcap/tidb/table/tables.(*partitionedTable).GetPartitionByRow\n\t/Users/pingcap/workspace/build-common@4/go/src/github.com/pingcap/tidb/table/tables/partition.go:1159\ngithub.com/pingcap/tidb/executor.(*dataReaderBuilder).buildTableReaderForIndexJoin\n\t/Users/pingcap/workspace/build-common@4/go/src/github.com/pingcap/tidb/executor/builder.go:4211\ngithub.com/pingcap/tidb/executor.(*dataReaderBuilder).buildExecutorForIndexJoinInternal\n\t/Users/pingcap/workspace/build-common@4/go/src/github.com/pingcap/tidb/executor/builder.go:4062\ngithub.com/pingcap/tidb/executor.(*dataReaderBuilder).buildExecutorForIndexJoin\n\t/Users/pingcap/workspace/build-common@4/go/src/github.com/pingcap/tidb/executor/builder.go:4055\ngithub.com/pingcap/tidb/executor.(*innerWorker).fetchInnerResults\n\t/Users/pingcap/workspace/build-common@4/go/src/github.com/pingcap/tidb/executor/index_lookup_join.go:690\ngithub.com/pingcap/tidb/executor.(*innerWorker).handleTask\n\t/Users/pingcap/workspace/build-common@4/go/src/github.com/pingcap/tidb/executor/index_lookup_join.go:528\ngithub.com/pingcap/tidb/executor.(*innerWorker).run\n\t/Users/pingcap/workspace/build-common@4/go/src/github.com/pingcap/tidb/executor/index_lookup_join.go:502
```

### 4. What is your TiDB version? (Required)
master

**Comment by wjhuang2016** [comment]
The hint may not work, and got 
``` sql
 desc select    /*+ inl_join( `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd` , `9aed23b9-9772-40dc-951c-7263c605c776` ) */ avg(   `9aed23b9-9772-40dc-951c-7263c605c776`.`c0083600-1ec7-4176-bbe2-b8a4a71596d7` ) as r0 , repeat( `9aed23b9-9772-40dc-951c-7263c605c776`.`90c28415-53f7-4443-b5e1-4177e42f0f39` , `9aed23b9-9772-40dc-951c-7263c605c776`.`c0083600-1ec7-4176-bbe2-b8a4a71596d7` ) as r1 , locate( `9aed23b9-9772-40dc-951c-7263c605c776`.`90c28415-53f7-4443-b5e1-4177e42f0f39` ,
`9aed23b9-9772-40dc-951c-7263c605c776`.`90c28415-53f7-4443-b5e1-4177e42f0f39` ) as r2 , unhex( `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd`.`791be5bd-3a41-434f-b194-0a8f7f5fa27b` ) as r3 from `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd` right join `9aed23b9-9772-40dc-951c-7263c605c776` on `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd`.`cb64194a-f141-4bfb-a2dd-dd78b6cf9f91` = `9aed23b9-9772-40dc-951c-7263c605c776`.`c0083600-1ec7-4176-bbe2-b8a4a71596d7` where not( `9aed23b9-9772-40dc-951c-7263c605c776`.`c0083600-1ec7-4176-bbe2-b8a4a71596d7` in ( -7860 ,-13384 ,-12940 ) ) and not( `ba2bc2d8-3932-44fa-8bd9-f7c6822020dd`.`cb64194a-f141-4bfb-a2dd-dd78b6cf9f91` between '4s7ht' and 'mj' );
+--------------------------------------+----------+-----------+----------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| id                                   | estRows  | task      | access object                                            | operator info                                                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
+--------------------------------------+----------+-----------+----------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Projection_15                        | 1.00     | root      |                                                          | Column#8, repeat(test1.9aed23b9-9772-40dc-951c-7263c605c776.90c28415-53f7-4443-b5e1-4177e42f0f39, test1.9aed23b9-9772-40dc-951c-7263c605c776.c0083600-1ec7-4176-bbe2-b8a4a71596d7)->Column#9, locate(test1.9aed23b9-9772-40dc-951c-7263c605c776.90c28415-53f7-4443-b5e1-4177e42f0f39, test1.9aed23b9-9772-40dc-951c-7263c605c776.90c28415-53f7-4443-b5e1-4177e42f0f39)->Column#10, unhex(cast(test1.ba2bc2d8-3932-44fa-8bd9-f7c6822020dd.791be5bd-3a41-434f-b194-0a8f7f5fa27b, var_string(5)))->Column#11 |
| └─StreamAgg_17                       | 1.00     | root      |                                                          | funcs:avg(Column#17)->Column#8, funcs:firstrow(Column#18)->test1.ba2bc2d8-3932-44fa-8bd9-f7c6822020dd.791be5bd-3a41-434f-b194-0a8f7f5fa27b, funcs:firstrow(Column#19)->test1.9aed23b9-9772-40dc-951c-7263c605c776.c0083600-1ec7-4176-bbe2-b8a4a71596d7, funcs:firstrow(Column#20)->test1.9aed23b9-9772-40dc-951c-7263c605c776.90c28415-53f7-4443-b5e1-4177e42f0f39                                                                                                                                        |
|   └─Projection_41                    | 7.00     | root      |                                                          | cast(test1.9aed23b9-9772-40dc-951c-7263c605c776.c0083600-1ec7-4176-bbe2-b8a4a71596d7, decimal(5,0) BINARY)->Column#17, test1.ba2bc2d8-3932-44fa-8bd9-f7c6822020dd.791be5bd-3a41-434f-b194-0a8f7f5fa27b, test1.9aed23b9-9772-40dc-951c-7263c605c776.c0083600-1ec7-4176-bbe2-b8a4a71596d7, test1.9aed23b9-9772-40dc-951c-7263c605c776.90c28415-53f7-4443-b5e1-4177e42f0f39                                                                                                                                  |
|     └─HashJoin_40                    | 7.00     | root      |                                                          | inner join, equal:[eq(Column#15, test1.9aed23b9-9772-40dc-951c-7263c605c776.c0083600-1ec7-4176-bbe2-b8a4a71596d7)]                                                                                                                                                                                                                                                                                                                                                                                        |
|       ├─Projection_20(Build)         | 5.60     | root      |                                                          | test1.ba2bc2d8-3932-44fa-8bd9-f7c6822020dd.791be5bd-3a41-434f-b194-0a8f7f5fa27b, cast(test1.ba2bc2d8-3932-44fa-8bd9-f7c6822020dd.cb64194a-f141-4bfb-a2dd-dd78b6cf9f91, bigint(39) BINARY)->Column#15                                                                                                                                                                                                                                                                                                      |
|       │ └─Selection_21               | 5.60     | root      |                                                          | not(and(ge(test1.ba2bc2d8-3932-44fa-8bd9-f7c6822020dd.cb64194a-f141-4bfb-a2dd-dd78b6cf9f91, "4s7ht"), le(test1.ba2bc2d8-3932-44fa-8bd9-f7c6822020dd.cb64194a-f141-4bfb-a2dd-dd78b6cf9f91, "mj")))                                                                                                                                                                                                                                                                                                         |
|       │   └─TableReader_23           | 7.00     | root      |                                                          | data:TableFullScan_22                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |
|       │     └─TableFullScan_22       | 7.00     | cop[tikv] | table:ba2bc2d8-3932-44fa-8bd9-f7c6822020dd               | keep order:false, stats:pseudo                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
|       └─PartitionUnion_30(Probe)     | 12250.41 | root      |                                                          |                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           |
|         ├─TableReader_32             | 7166.67  | root      |                                                          | data:TableRangeScan_31                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
|         │ └─TableRangeScan_31        | 7166.67  | cop[tikv] | table:9aed23b9-9772-40dc-951c-7263c605c776, partition:p0 | range:[-inf,-13384), (-13384,-12940), (-12940,-7860), (-7860,+inf], keep order:false, stats:pseudo                                                                                                                                                                                                                                                                                                                                                                                                        |
|         ├─TableReader_34             | 7166.67  | root      |                                                          | data:TableRangeScan_33                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
|         │ └─TableRangeScan_33        | 7166.67  | cop[tikv] | table:9aed23b9-9772-40dc-951c-7263c605c776, partition:p1 | range:[-inf,-13384), (-13384,-12940), (-12940,-7860), (-7860,+inf], keep order:false, stats:pseudo                                                                                                                                                                                                                                                                                                                                                                                                        |
|         ├─TableReader_36             | 1.00     | root      |                                                          | data:TableRangeScan_35                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
|         │ └─TableRangeScan_35        | 1.00     | cop[tikv] | table:9aed23b9-9772-40dc-951c-7263c605c776, partition:p2 | range:[-inf,-13384), (-13384,-12940), (-12940,-7860), (-7860,+inf], keep order:false, stats:pseudo                                                                                                                                                                                                                                                                                                                                                                                                        |
|         └─TableReader_38             | 7166.67  | root      |                                                          | data:TableRangeScan_37                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
|           └─TableRangeScan_37        | 7166.67  | cop[tikv] | table:9aed23b9-9772-40dc-951c-7263c605c776, partition:p3 | range:[-inf,-13384), (-13384,-12940), (-12940,-7860), (-7860,+inf], keep order:false, stats:pseudo                                                                                                                                                                                                                                                                                                                                                                                                        |
+--------------------------------------+----------+-----------+----------------------------------------------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
```

If so, please analyze these two tables.

**Comment by crazycs520** [comment]
The reproduce can be simply to:

```sql
CREATE TABLE t1 (
  c1 double DEFAULT '1.335088259490289',
  c2 set('mj','4s7ht','z','3i','b26','9','cg11','uvzcp','c','ns','fl9') NOT NULL DEFAULT 'mj,z,3i,9,cg11,c',
  PRIMARY KEY (c2) /*T![clustered_index] CLUSTERED */,
  KEY i1 (c1),
  KEY i2 (c1),
  KEY i3 (c1)
) ENGINE=InnoDB DEFAULT CHARSET=gbk COLLATE=gbk_chinese_ci;

INSERT INTO t1 VALUES (634.2783557491367,''),(2000.5041449792013,'4s7ht'),(634.2783557491367,'3i'),(634.2783557491367,'9'),(7803.173688589342,'uvzcp'),(634.2783557491367,'ns'),(634.2783557491367,'fl9');

CREATE TABLE t2 (
  c3 decimal(56,16) DEFAULT '931359772706767457132645278260455518957.9866038319986886',
  c4 set('3bqx','g','6op3','2g','jf','arkd3','y0b','jdy','1g','ff5z','224b') DEFAULT '3bqx,2g,ff5z,224b',
  c5 smallint(6) NOT NULL DEFAULT '-25973',
  c6 year(4) DEFAULT '2122',
  c7 text DEFAULT NULL,
  PRIMARY KEY (c5) /*T![clustered_index] CLUSTERED */,
  KEY i4 (c6),
  KEY i5 (c5)
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_bin COMMENT=''
PARTITION BY HASH (c5) PARTITIONS 4;

INSERT INTO t2 VALUES (465.0000000000000000,'jdy',-8542,2008,'FgZXe');

analyze table t1;
analyze table t2;
set @@sql_mode='';

select    /*+ inl_join( t1 , t2 ) */ avg(   t2.c5 ) as r0 , repeat( t2.c7 , t2.c5 ) as r1 , locate( t2.c7 , t2.c7 ) as r2 , unhex( t1.c1 ) as r3 from t1 right join t2 on t1.c2 = t2.c5 where not( t2.c5 in ( -7860 ,-13384 ,-12940 ) ) and not( t1.c2 between '4s7ht' and 'mj' );

```

**Comment by crazycs520** [comment]
The reason is in `tidb/executor/builder.go:4211`,  The input parameters of the API `GetPartitionByRow` are incorrect, the second parameter should be a **row** `[]types.Datum` which contain all columns `Datum` value, instead of currently only contains the partial column values. 


<img width="846" alt="image" src="https://user-images.githubusercontent.com/26020263/212845203-88c049ff-4d93-4d33-a981-5654021ab5c4.png">

**Comment by tiancaiamao** [comment]
```
mysql> explain select    /*+ inl_join( t1 , t2 ) */ avg(   t2.c5 ) as r0 , repeat( t2.c7 , t2.c5 ) as r1 , locate( t2.c7 , t2.c7 ) as r2 , unhex( t1.c1 ) as r3 from t1 right join t2 on t1.c2 = t2.c5 where not
( t2.c5 in ( -7860 ,-13384 ,-12940 ) ) and not( t1.c2 between '4s7ht' and 'mj' );
+--------------------------------------+----------+-----------+------------------------+------------------------------------------------------------------------------------------------------------------------
-----------------------------+
| id                                   | estRows  | task      | access object          | operator info
                             |
+--------------------------------------+----------+-----------+------------------------+------------------------------------------------------------------------------------------------------------------------
-----------------------------+
| Projection_15                        | 1.00     | root      |                        | Column#9, repeat(test.t2.c7, test.t2.c5)->Column#10, locate(test.t2.c7, test.t2.c7)->Column#11, unhex(cast(test.t1.c1,
var_string(5)))->Column#12   |
| └─HashAgg_16                         | 1.00     | root      |                        | funcs:avg(Column#18)->Column#9, funcs:firstrow(Column#19)->test.t1.c1, funcs:firstrow(Column#20)->test.t2.c5, funcs:fir
strow(Column#21)->test.t2.c7 |
|   └─Projection_35                    | 14.00    | root      |                        | cast(test.t2.c5, decimal(5,0) BINARY)->Column#18, test.t1.c1, test.t2.c5, test.t2.c7
                             |
|     └─HashJoin_19                    | 14.00    | root      |                        | inner join, equal:[eq(Column#16, test.t2.c5)]
                             |
|       ├─Projection_20(Build)         | 11.20    | root      |                        | test.t1.c1, cast(test.t1.c2, bigint(39) BINARY)->Column#16
                             |
|       │ └─Selection_21               | 11.20    | root      |                        | not(and(ge(test.t1.c2, "4s7ht"), le(test.t1.c2, "mj")))
                             |
|       │   └─TableReader_23           | 14.00    | root      |                        | data:TableFullScan_22
                             |
|       │     └─TableFullScan_22       | 14.00    | cop[tikv] | table:t1               | keep order:false
                             |
|       └─PartitionUnion_24(Probe)     | 12250.82 | root      |                        |
                             |
|         ├─TableReader_26             | 7166.67  | root      |                        | data:TableRangeScan_25
                             |
|         │ └─TableRangeScan_25        | 7166.67  | cop[tikv] | table:t2, partition:p0 | range:[-inf,-13384), (-13384,-12940), (-12940,-7860), (-7860,+inf], keep order:false, stats:pseudo
                             |
|         ├─TableReader_28             | 7166.67  | root      |                        | data:TableRangeScan_27
                             |
|         │ └─TableRangeScan_27        | 7166.67  | cop[tikv] | table:t2, partition:p1 | range:[-inf,-13384), (-13384,-12940), (-12940,-7860), (-7860,+inf], keep order:false, stats:pseudo
                             |
|         ├─TableReader_30             | 1.43     | root      |                        | data:TableRangeScan_29
                             |
|         │ └─TableRangeScan_29        | 1.43     | cop[tikv] | table:t2, partition:p2 | range:[-inf,-13384), (-13384,-12940), (-12940,-7860), (-7860,+inf], keep order:false, stats:pseudo
                             |
|         └─TableReader_32             | 7166.67  | root      |                        | data:TableRangeScan_31
                             |
|           └─TableRangeScan_31        | 7166.67  | cop[tikv] | table:t2, partition:p3 | range:[-inf,-13384), (-13384,-12940), (-12940,-7860), (-7860,+inf], keep order:false, stats:pseudo
                             |
+--------------------------------------+----------+-----------+------------------------+------------------------------------------------------------------------------------------------------------------------
-----------------------------+
17 rows in set, 2 warnings (0.00 sec)

mysql> show warnings;
+---------+------+------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                            |
+---------+------+------------------------------------------------------------------------------------+
| Warning | 1815 | Optimizer Hint /*+ INL_JOIN(t1, t2) */ or /*+ TIDB_INLJ(t1, t2) */ is inapplicable |
| Warning | 1815 | Optimizer Hint /*+ INL_JOIN(t1, t2) */ or /*+ TIDB_INLJ(t1, t2) */ is inapplicable |
+---------+------+------------------------------------------------------------------------------------+
2 rows in set (0.00 sec)

mysql> select tidb_version();
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-----------------------------------------------------------------------------------------+
| tidb_version()
                                                                                         |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-----------------------------------------------------------------------------------------+
| Release Version: v6.6.0-alpha-245-gf7d35abd12
Edition: Community
Git Commit Hash: f7d35abd12ab0197cb6c3e24086dc075b9425f73
Git Branch: master
UTC Build Time: 2023-01-17 12:09:26
GoVersion: go1.19.4
Race Enabled: false
TiKV Min Version: 6.2.0-alpha
Check Table Before Drop: false
Store: unistore |
+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-----------------------------------------------------------------------------------------+
1 row in set (0.00 sec)
```

There is a warning, index join is not choosed.

**Comment by tiancaiamao** [comment]
A minimal reproduce query:

```
select /*+ inl_join (t1, t2) */ t2.c5
from t1 right join t2
on t1.c2 = t2.c5
where not( t1.c2 between '4s7ht' and 'mj' );
```

**Comment by mjonss** [comment]
This fix should also be included in 6.1, see https://github.com/pingcap/tidb/issues/40803.
